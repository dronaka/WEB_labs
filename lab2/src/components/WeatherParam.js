import React from "react";



export default function WeatherParam(props) {
  return (
    <div className="param">
      <div className="name">{props.name}</div>
      <div className="value">{props.value}</div>
    </div>
  );
}