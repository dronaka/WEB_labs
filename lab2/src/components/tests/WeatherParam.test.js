import React from "react";
import renderer from "react-test-renderer";

import WeatherParam from "../WeatherParam";


describe("<WeatherParam /> render", () =>{

  const DEFAULT_PROPS = {
    name: "Temperature",
    value: "25",
  };

  it("should has name and value", () => {
    const { name, value } = DEFAULT_PROPS;
    
    const tree = renderer
      .create(
        <WeatherParam
          name={name}
          value={value}
        />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});