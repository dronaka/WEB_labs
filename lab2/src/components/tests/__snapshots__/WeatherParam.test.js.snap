// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherParam /> render should has name and value 1`] = `
<div
  className="param"
>
  <div
    className="name"
  >
    Temperature
  </div>
  <div
    className="value"
  >
    25
  </div>
</div>
`;
